packages/react/time-series/src/TimeSeries.tsx

```diff
@@ -257,15 +259,22 @@ export function TimeSeries(props: TimeSeriesProps) {
     try {
       setIsLoading(true)
 
+      const filters = JSON.parse(filtersString)
```

Comment:
same transformation here. Maybe I'm missing something?

```suggestion
      const filters = query?.filters || []
```

Reply:
The reason for this is that we cannot pass `filters` directly in the dependency array, since `filters` is an array that is not necessarily memoized, it would cause the `useEffect` to trigger on every re-render.

A solution for this is transforming the array to a string, which can be compared  using `===`.

Another solution would be making a deep comparison using a library like [use-deep-compare-effect](https://github.com/kentcdodds/use-deep-compare-effect) but we'd need to add a new dependency and deep comparisons may end up being very costly.

https://github.com/facebook/react/issues/14476#issuecomment-471199055